kvclientpool.go

```diff
@@ -439,7 +477,51 @@ func (p *kvClientPool) getClientSlow(ctx context.Context) (KvClient, error) {
 func (p *kvClientPool) Shutdown(ctx context.Context) {
 }
 
-func (p *kvClientPool) Close() {
+func (p *kvClientPool) Close() error {
+	select {
+	case <-p.closed:
+		return nil
+	default:
+	}
+
+	p.logger.Info("Closing kv client pool")
+	close(p.closed)
+	p.lock.Lock()
+
+	p.fastMap.Store(nil)
+
+	// Close anyone waiting on this.
+	if p.needNoDefunctSigCh != nil {
+		close(p.needNoDefunctSigCh)
+		p.needNoDefunctSigCh = nil
+	}
+
+	// Just close all the clients in all the lists containing alive clients.
+	for _, client := range p.activeClients {
```

Comment:
activeClients is more or less meant to be a 'cache' rather than something that is actively operated on.  It might make sense in that context to separately iterate the currentClients and defunctClients, and then update the 'activeClients cache' afterwards by clearing it.